### DISCUSSION TITLE
I2S on STM32U5xx

### DISCUSSION THREAD

**Discussion by chrisschir** [zephyrproject-rtos/zephyr#75565] [Q&A] [answered]
I want to use I2S on the STM32U585. I have tried to use (and modify) the existing drivers, but I was not successful.
If I compare the reference manuals of the U5 and H7 it seems to me that the U5 has a different architecture for I2S. On the H7 I2S is inside the SPI unit and the SAI unit; on the U5 it is only in the SAI unit. 

Can I use the i2s_ll_stm32 for the STM32U5xx? Or is the architecture of the U5 different that I can not use that driver?

**Comment by mariopaja** [comment, marked as answer]
Hi @chrisschir,
At the moment there is no SAI support for STM32U5

> Can I use the i2s_ll_stm32 for the STM32U5xx?

No you cannot. The actual implementation is only for MCUs that have a muxed I2S and SPI https://github.com/zephyrproject-rtos/zephyr/pull/72154#issuecomment-2098355638
